agent/config/builder.go

```diff
@@ -2581,11 +2582,37 @@ func validateAutoConfigAuthorizer(rt RuntimeConfig) error {
 }
 
 func (b *builder) cloudConfigVal(v Config) hcpconfig.CloudConfig {
+	// Load the same environment variables expected by hcp-sdk-go
+	envHostname, ok := os.LookupEnv("HCP_API_ADDRESS")
+	if !ok {
+		if legacyEnvHostname, ok := os.LookupEnv("HCP_API_HOST"); ok {
+			// Remove only https scheme prefixes from the deprecated environment
+			// variable for specifying the API host. Mirrors the same behavior as
+			// hcp-sdk-go.
+			if strings.HasPrefix(strings.ToLower(legacyEnvHostname), "https://") {
+				legacyEnvHostname = legacyEnvHostname[8:]
+			}
+			envHostname = legacyEnvHostname
+		}
+	}
+
+	var envTLSConfig *tls.Config
+	if os.Getenv("HCP_AUTH_TLS") == "insecure" ||
+		os.Getenv("HCP_SCADA_TLS") == "insecure" ||
+		os.Getenv("HCP_API_TLS") == "insecure" {
+		envTLSConfig = &tls.Config{InsecureSkipVerify: true}
+	}
```

Comment:
Is there any way we can use the hcp-sdk-go config loader here directly instead of re-implementing this behavior?

Reply:
Hmm, looks like that change would be a bit more extensive since `NewHCPConfig` returns `(hcpcfg.HCPConfig, error)` and this function shouldn't ever return an error. Is that partly why you re-implemented this? I think it makes sense 👍

Reply:
Yeah, it was that plus `NewHCPConfig` does some validation at the end, which would make complicate things for scenarios like `client_id` set in the config file and `HCP_CLIENT_SECRET` set in the environment variables. If I just used `FromEnv`, then the validation would fail because the client ID isn't set.

So I'd have to either also set some fake config values or load the file values too, which seemed overly complicated and felt out of scope for a config builder method! Maybe there's a way to enhance this in the future, though, with a change to the hcp-sdk-go library.